libs/core/tests/unit_tests/test_tools.py

```diff
@@ -2174,3 +2175,124 @@ def foo(x: int) -> Bar:
     assert foo.invoke(
         {"type": "tool_call", "args": {"x": 0}, "name": "foo", "id": "bar"}
     ) == Bar(x=0)
+
+
+def test_method_tool_self_ref() -> None:
+    """Test that a method tool can reference self."""
+
+    class A:
+        def __init__(self, c: int):
+            self.c = c
+
+        @MethodTool
```

Comment:
```suggestion
        @methodtool
```
or `toolmethod` - @baskaryan do you have a preference?

Reply:
Was getting linter errors for this, which will not be a problem if changing the decorator to a function rather than a class is required.